File: C#/4. PizzaStore/Classes/Pizza.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PizzaStore.Classes
{
    public abstract class Pizza
    {
        protected string _Name;
        protected string _Dough;
        protected string _Sauce;
        protected List<string> _Toppings = new List<string>();

        public void Prepare()
        {
            Console.WriteLine("Preparing " + _Name);
            Console.WriteLine("Tossing dough...");
            Console.WriteLine("Adding sauce...");
            Console.WriteLine("Adding toppings:");
            foreach (string topping in _Toppings)
            {
                Console.WriteLine(" " + topping);
            }
        }

        public void Bake()
        {
            Console.WriteLine("Bake for 25 minutos at 350");
        }

        public virtual void Cut()
        {
            Console.WriteLine("Cutting the pizza into diagonal slices");
        }

        public void Box()
        {
            Console.WriteLine("Place pizza in official PizzaStore box");
        }

        public string Name
        {
            get { return _Name; }
        }
    }
}

File: Java/4-PizzaStoreFactory/src/main/java/com/mycompany/pizzastorefactory/Pizza.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.pizzastorefactory;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author ismas
 */
public abstract class Pizza {
    protected String name;
    protected String dough;
    protected String sauce;
    protected List<String> toppings = new ArrayList<String>();
    
    void prepare(){
        System.out.println("Preparing " + name);
        System.out.println("Tossing dough...");
        System.out.println("Adding sauce...");
        System.out.println("Adding toppings: ");
        for( String topping : toppings ){
            System.out.println("   " + topping);
        }
    }
    
    void bake(){
        System.out.println("Bake for 25 minutes at 350");
    }
    
    protected void cut() {
        System.out.println("Cutting the pizza into diagonal slices");
    }
    
    void box() {
        System.out.println("Place pizza in official PizzaStore box");
    }
    
    public String getName() {
        return name;
    }
}
